cmd/src/debug_comp.go

```diff
@@ -0,0 +1,270 @@
+
+	// start goroutine to run docker ps -o wide
+	wg.Add(1)
+	go func() {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return
+		}
+		defer semaphore.Release(1)
+		defer wg.Done()
+		ch <- getPs(ctx, baseDir)
+	}()
+
+	// start goroutine to run docker container stats --no-stream
+	wg.Add(1)
+	go func() {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return
+		}
+		defer semaphore.Release(1)
+		defer wg.Done()
+		ch <- getStats(ctx, baseDir)
+	}()
+
+	// start goroutine to run docker container logs <container>
+	for _, container := range containers {
+		wg.Add(1)
+		go func(container string) {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getContainerLog(ctx, container, baseDir)
+		}(container)
+	}
+
+	// start goroutine to run docker container inspect <container>
+	for _, container := range containers {
+		wg.Add(1)
+		go func(container string) {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getInspect(ctx, container, baseDir)
+		}(container)
+	}
+
+	// start goroutine to get configs
+	if configs {
+		wg.Add(1)
+		go func() {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getExternalServicesConfig(ctx, baseDir)
+		}()
+
+		wg.Add(1)
+		go func() {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getSiteConfig(ctx, baseDir)
+		}()
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		wg.Wait()
+		close(ch)
+	}()
+
+	for f := range ch {
+		if f.err != nil {
+			return fmt.Errorf("aborting due to error on %s: %v\noutput: %s", f.name, f.err, f.data)
+		}
+
+		if verbose {
+			log.Printf("archiving file %q with %d bytes", f.name, len(f.data))
+		}
+
+		zf, err := zw.Create(f.name)
+		if err != nil {
+			return fmt.Errorf("failed to create %s: %w", f.name, err)
+		}
+
+		_, err = zf.Write(f.data)
+		if err != nil {
+			return fmt.Errorf("failed to write to %s: %w", f.name, err)
+		}
+	}
+
+	return nil
+}
+
+func getContainers(ctx context.Context) ([]string, error) {
+	c, err := exec.CommandContext(ctx, "docker", "container", "ls", "--format", "{{.Names}} {{.Networks}}").Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get container names with error: %w", err)
+	}
+	s := string(c)
+	preprocessed := strings.Split(strings.TrimSpace(s), "\n")
+	containers := []string{}
+	for _, container := range preprocessed {
+		tmpStr := strings.Split(container, " ")
+		if tmpStr[1] == "docker-compose_sourcegraph" {
+			containers = append(containers, tmpStr[0])
+		}
+	}
+	return containers, err
+}
+
+func getPs(ctx context.Context, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/docker/docker-ps")+".txt",
```

Comment:
```suggestion
		filepath.Join(baseDir, "docker", "docker-ps.txt),
```

When using `filepath.Join` you'd like to break apart all of the path components so that you don't need to use a literal path sep in the strings you supply. (They can be different on different OSes)

Reply:
Fixed ✅